src/test_l1_dsAudio.c

```diff
@@ -3485,41 +4440,52 @@ void test_l1_dsAudio_positive_dsGetMS12AudioProfile(void) {
 	gTestID = 55;
 	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
-	dsError_t result;
-	intptr_t handle[NUM_OF_PORTS] = {NUM_OF_PORTS};
-	char currentProfile[MAX_PROFILE_NAME_LEN];
+        dsError_t result;
+        intptr_t* handle = NULL;
+        char currentProfile[MAX_PROFILE_NAME_LEN];
 
-	// Step 01: Initialize audio ports
-	result = dsAudioPortInit();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // allocate memory for the handles
+        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+        if(handle == NULL)
+        {
+                UT_LOG_ERROR("Failed to allcoate memory for handlers");
+                return;
+        }
 
-	// Step 02: Get the port handle for all supported audio ports
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+        // Step 01: Initialize audio ports
+        result = dsAudioPortInit();
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-		if(handle[i] == dsAUDIOPORT_TYPE_SPEAKER) {
-			// Step 03: Get the current MS12 audio profile for Speaker. MS12 Audio Profile is supported only on SPEAKER Port
-			result = dsGetMS12AudioProfile(handle[i], currentProfile);
-			UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // Step 02: Get the port handle for all supported audio ports
+        for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
-			// Step 04: Verify that the profile string is valid and not empty
-			UT_ASSERT_NOT_EQUAL(strlen(currentProfile), 0);
-		}
+                if(handle[i] == dsAUDIOPORT_TYPE_SPEAKER) {
```

Comment:
comparing handle with type is not correct

Reply:
done